aci/resource_aci_fvesg.go

```diff
@@ -0,0 +1,696 @@
+
+	if fvESg.DistinguishedName == "" {
+		return nil, fmt.Errorf("EndpointSecurityGroup %s not found", fvESg.DistinguishedName)
+	}
+
+	return fvESg, nil
+}
+
+func setEndpointSecurityGroupAttributes(fvESg *models.EndpointSecurityGroup, d *schema.ResourceData) *schema.ResourceData {
+	d.SetId(fvESg.DistinguishedName)
+	d.Set("description", fvESg.Description)
+	fvESgMap, _ := fvESg.ToMap()
+
+	d.Set("name", fvESgMap["name"])
+	d.Set("application_profile_dn", GetParentDn(fvESg.DistinguishedName, fmt.Sprintf("/esg-%s", fvESgMap["name"])))
+
+	d.Set("annotation", fvESgMap["annotation"])
+	d.Set("exception_tag", fvESgMap["exceptionTag"])
+	d.Set("flood_on_encap", fvESgMap["floodOnEncap"])
+	d.Set("match_t", fvESgMap["matchT"])
+	d.Set("name_alias", fvESgMap["nameAlias"])
+	d.Set("pc_enf_pref", fvESgMap["pcEnfPref"])
+	d.Set("pref_gr_memb", fvESgMap["prefGrMemb"])
+	d.Set("prio", fvESgMap["prio"])
+	d.Set("userdom", fvESgMap["userdom"])
+	return d
+}
+
+func resourceAciEndpointSecurityGroupImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] %s: Beginning Import", d.Id())
+	aciClient := m.(*client.Client)
+
+	dn := d.Id()
+
+	fvESg, err := getRemoteEndpointSecurityGroup(aciClient, dn)
+
+	if err != nil {
+		return nil, err
+	}
+	schemaFilled := setEndpointSecurityGroupAttributes(fvESg, d)
+
+	log.Printf("[DEBUG] %s: Import finished successfully", d.Id())
+
+	return []*schema.ResourceData{schemaFilled}, nil
+}
+
+func resourceAciEndpointSecurityGroupCreate(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] EndpointSecurityGroup: Beginning Creation")
+	aciClient := m.(*client.Client)
+	desc := d.Get("description").(string)
+
+	name := d.Get("name").(string)
+
+	ApplicationProfileDn := d.Get("application_profile_dn").(string)
+
+	fvESgAttr := models.EndpointSecurityGroupAttributes{}
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		fvESgAttr.Annotation = Annotation.(string)
+	}
+	if ExceptionTag, ok := d.GetOk("exception_tag"); ok {
+		fvESgAttr.ExceptionTag = ExceptionTag.(string)
+	}
+	if FloodOnEncap, ok := d.GetOk("flood_on_encap"); ok {
+		fvESgAttr.FloodOnEncap = FloodOnEncap.(string)
+	}
+	if MatchT, ok := d.GetOk("match_t"); ok {
+		fvESgAttr.MatchT = MatchT.(string)
+	}
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		fvESgAttr.NameAlias = NameAlias.(string)
+	}
+	if PcEnfPref, ok := d.GetOk("pc_enf_pref"); ok {
+		fvESgAttr.PcEnfPref = PcEnfPref.(string)
+	}
+	if PrefGrMemb, ok := d.GetOk("pref_gr_memb"); ok {
+		fvESgAttr.PrefGrMemb = PrefGrMemb.(string)
+	}
+	if Prio, ok := d.GetOk("prio"); ok {
+		fvESgAttr.Prio = Prio.(string)
+	}
+	if Userdom, ok := d.GetOk("userdom"); ok {
+		fvESgAttr.Userdom = Userdom.(string)
+	}
+	fvESg := models.NewEndpointSecurityGroup(fmt.Sprintf("esg-%s", name), ApplicationProfileDn, desc, fvESgAttr)
+
+	err := aciClient.Save(fvESg)
+	if err != nil {
+		return err
+	}
+	d.Partial(true)
+
+	d.SetPartial("name")
+
+	d.Partial(false)
```

Comment:
There should one pre check for all relations. That will check whether relation DN actually exist or not.